agent/csec_metadata.c

```diff
@@ -0,0 +1,48 @@
+#include "csec_metadata.h"
+#include "util_strings.h"
+#include "php_hash.h"
+#include "php_api_internal.h"
+
+static void nr_csec_php_add_assoc_string_const(zval* arr,
+                                          const char* key,
+                                          const char* value) {
+  char* val = NULL;
+
+  if (NULL == arr || NULL == key || NULL == value) {
+    return;
+  }
+
+  val = nr_strdup(value);
+  nr_php_add_assoc_string(arr, key, val);
+  nr_free(val);
+}
+
+#ifdef TAGS
+void zif_newrelic_get_security_metadata(void); /* ctags landing pad only */
+void newrelic_get_security_metadata(void);     /* ctags landing pad only */
+#endif
+PHP_FUNCTION(newrelic_get_security_metadata) {
```

Comment:
Does this function need to be called from PHP?

Reply:
No but this function will be called by the Security Agent

Reply:
If that's the case then this shouldn't be a PHP user function but a function exported by newrelic-php-agent shared object, callable from a program by the means of dynamic binding - see https://linux.die.net/man/3/dlsym.

Reply:
newrelic-php-agent already does similar thing - see [here](https://github.com/newrelic/newrelic-php-agent/blob/R10.21/agent/php_minit.c#L387).

Reply:
Thanks Michal, I will update the code and not expose this as a PHP user function. Thanks for the reference too.

Reply:
I tried the dynamic call using dlopen and dlsym, but seems like the values may not be accessible.
`nr_app_get_entity_name(NRPRG(app))` is one of the calls which is required, but `NRPRG` is the zend module's address space which does not seem accessible directly.

`fprintf(stderr, "%p : newrelic_globals.app\n", NRPRG(app));`

gives `(nil)` when called from dlopen'ed binary while from inside a valid address is printed.

Lets connect over call to finalise a solution.

Reply:
@anmol-ap Please take a look at #922 - it should have what's needed.

Reply:
Yes, the special purpose value for handle worked. Thanks.